packages/core/src/components/tooltip/tooltip.tsx

```diff
@@ -114,13 +115,36 @@ export class Tooltip<
         this.popoverRef.current?.reposition();
     }
 
+    protected validateProps(props: TooltipProps<T>) {
+        const childrenCount = React.Children.count(props.children);
+        if (childrenCount > 1) {
+            console.warn(Errors.POPOVER_WARN_TOO_MANY_CHILDREN);
+        }
+        // all other warnings should occur in Popover, not here.
+    }
+
     // any descendant ContextMenus may update this ctxState
     private renderPopover = (ctxState: TooltipContextState) => {
-        const { children, compact, disabled, intent, popoverClassName, ...restProps } = this.props;
+        const { children, content, renderTarget, compact, disabled, intent, popoverClassName, ...restProps } =
+            this.props;
+
         const popoverClasses = classNames(Classes.TOOLTIP, Classes.intentClass(intent), popoverClassName, {
             [Classes.COMPACT]: compact,
         });
 
+        const contentElement = Utils.ensureElement(content);
+        const tooltipId = contentElement?.props?.id ?? Utils.uniqueId("tooltip");
+        const clonedContent = contentElement
+            ? React.cloneElement(contentElement, { role: "tooltip", id: tooltipId })
+            : contentElement;
+
+        const childTarget = Utils.ensureElement(React.Children.toArray(children)[0]);
+        const clonedTarget = childTarget
+            ? React.cloneElement(childTarget, {
+                  "aria-describedby": [tooltipId, childTarget.props["aria-describedby"]].filter(Boolean).join(" "),
```

Comment:
`tooltipId` could possibly appear here twice in cases where consumers of `Tooltip` have already added these `id`/`aria-describedby` attributes

I'm not sure of the implications of that but it seems easy enough to guard against

Reply:
I don't think that would make a difference in a11y but here: https://github.com/palantir/blueprint/pull/6865/commits/05113cbe6229d2c885854afd0745b8ed148bdaff

Reply:
Actually here: https://github.com/palantir/blueprint/pull/6865/commits/be533a0f056508aa04fbc46bf71dd3f797de080f